src/app/zwiazek/kalendarz/_components/CalendarTable.tsx

```diff
@@ -0,0 +1,103 @@
+  nextMonth,
+}: CalendarTableProps) => {
+  const generateCalendar = (date: Date) => {
+    const year = date.getFullYear();
+    const month = date.getMonth();
+    const currentDate = new Date(year, month, 1);
+    const calendarDays = [];
+    const dayOfWeek = currentDate.getDay(); // Numer dnia tygodnia
+    const daysToMonday = dayOfWeek === 0 ? -6 : 1 - dayOfWeek;
+    currentDate.setDate(currentDate.getDate() + daysToMonday);
+
+    for (let i = 0; i < 6; i++) {
+      const week = [];
+      for (let j = 0; j < 7; j++) {
+        week.push(formatDate(currentDate.toString()));
+        currentDate.setDate(currentDate.getDate() + 1);
+      }
+      calendarDays.push(week);
+    }
+
+    return calendarDays;
+  };
+
+  const calendarDays = generateCalendar(currentDate);
+
+  const eventsDates = cardsPlaceholder.map((card) => card.date);
+
+  return (
+    <div className="flex flex-col items-center gap-5 font-lora tablet:rounded tablet:border-2 tablet:border-greenC tablet:px-16 tablet:py-10">
+      <CalendarHeader
+        className="w-full items-center justify-between px-1.5 text-24 text-white tablet:text-32"
+        iconClassName="stroke-white tablet:w-8 tablet:h-8 "
+        currentDate={currentDate}
+        handlePrevMonth={prevMonth}
+        handleNextMonth={nextMonth}
+      />
+      <table className="flex flex-col items-center gap-5">
+        <thead>
+          <tr className="flex items-center border-b-2 border-greenC text-greenC tablet:gap-6">
+            {daysOfWeek.map((day) => (
+              <td
+                key={day}
+                aria-label={day}
+                className="h-10 w-10 text-center text-18 tablet:w-11 tablet:text-24"
+              >
+                {day}
+              </td>
+            ))}
+          </tr>
+        </thead>
+        <tbody className="text-center">
+          {calendarDays.map((week, weekIndex) => (
+            <tr
+              key={weekIndex}
+              className="flex items-center gap-0.5 tablet:gap-6"
+            >
+              {week.map((day, dayIndex) => (
+                <td
+                  key={dayIndex}
+                  aria-label={day}
+                  className={clsx(
+                    `flex h-10 w-10 items-center justify-center text-18 tablet:my-3 tablet:w-11 tablet:text-32`,
+                    day.split(".")[1] ===
+                      formatDate(currentDate.toString()).split(".")[1]
+                      ? "text-white"
+                      : "text-greenC",
+                    day === formatDate(today.toString()) &&
+                      "rounded border-2 border-yellowVintage",
+                    eventsDates.find(
+                      (date) =>
+                        date.split(".")[0] === day.split(".")[0] &&
+                        date.split(".")[1] === day.split(".")[1],
+                    ) && "rounded border-2 border-backgroundMain",
+                  )}
+                >
+                  {day.split(".")[0]}
```

Comment:
A lot of extraction of parts of the date:`split(".")[index]`
Just create a read model once. Transform the data you receive from the source (API/mock/static) to the model you want to use within your features. If it is only static data then you can model it straight away to fit the application data model.